pithos/plugins/notification_icon.py

```diff
@@ -112,6 +79,18 @@ def check_for_plasma():
                 None
             )
 
+        def on_StatusNotifierWatcher_finish(source, result, data):
+            try:
+                StatusNotifierWatcher = Gio.DBusProxy.new_finish(result)
+            except GLib.Error as e:
+                logging.debug(e)
+                check_for_x()
+            else:
+                if StatusNotifierWatcher.get_name_owner():
+                    init_appindicator()
+                else:
+                    check_for_plasma()
```

Comment:
> Yea, at a glance maybe it is more work than its worth.

Maybe but here is a starting point I think. https://freedesktop.org/wiki/Specifications/StatusNotifierItem/

If I can decipher MPRIS I can prop figure this out. Just might take a while.